ext/semian/resource.c

```diff
@@ -53,6 +60,21 @@ semian_resource_acquire(int argc, VALUE *argv, VALUE self)
     rb_raise(rb_eArgError, "invalid arguments");
   }
 
+  // Backup the timeout values in case we need to revert them
+  tv_sec = res.timeout.tv_sec;
+  tv_nsec = res.timeout.tv_nsec;
+
+  // If we recently booted, and are using the quota strategy
+  // we need to give the workers some time to register or else
+  // we may have spurious timeouts due to a small number of registered workers.
+  if(res.quota > 0) {
+    semaphore_stat(res.sem_id, &sem_ds);
+    if (difftime(time(0), sem_ds.sem_ctime) < QUOTA_BOOT_GRACE_PERIOD) {
+      res.timeout.tv_sec = res.quota_grace_timeout;
+      res.timeout.tv_nsec = 0;
```

Comment:
Maybe we should use the `ms_to_timespec` helper to support doubles? I think right now that wouldn't work, e.g. if you passed `55.5`, it'd probably just do `56` since `tv_sec` probably has to be an `int`?